source/common/upstream/subset_lb.cc

```diff
@@ -355,13 +373,23 @@ SubsetLoadBalancer::extractSubsetMetadata(const std::set<std::string>& subset_ke
   for (const auto key : subset_keys) {
     const auto it = fields.find(key);
     if (it == fields.end()) {
+      kvs.clear();
       break;
     }
-    kvs.emplace_back(std::pair<std::string, ProtobufWkt::Value>(key, it->second));
-  }
 
-  if (kvs.size() != subset_keys.size()) {
-    kvs.clear();
+    if (list_as_any_) {
+      // TODO(snowp): Should we switch on the the type of the oneof instead? This might add an empty
+      // list.
```

Comment:
Seems like you avoid the ambiguity by reverting to the original implementation and just handling lists specially in `findOrCreateSubset`? I'll add a note there with an alternate implementation.